tools/gulp/tasks/lint.ts

```diff
@@ -46,6 +54,56 @@ task('madge', ['material:clean-build'], () => {
   });
 });
 
+task('ownerslint', () => {
+  let errors = 0;
+
+  let ownedPaths = readFileSync(ownersFilePath, 'utf8').split('\n')
+      // Trim lines.
+      .map(line => line.trim())
+      // Remove empty lines and comments.
+      .filter(line => line && !line.startsWith('#'))
+      // Split off just the path glob and turn it into a regex.
+      .map(line => new Minimatch(line.split(/\s+/)[0], {dot: true, matchBase: true}))
+      // Add in the paths we're ignoring.
+      .concat(ignoreOwners.map(path => new Minimatch(path, {dot: true, matchBase: true})));
+
+  for (let paths = getChildPaths('.'); paths.length;) {
+    paths = Array.prototype.concat(...paths
+        // Remove paths that match an owned path.
+        .filter(path => !ownedPaths.reduce(
+            (isOwned, ownedPath) => isOwned || isOwnedBy(path, ownedPath), false))
+        // Report an error for any files that didn't match any owned paths.
+        .filter(path => {
+          if (statSync(path).isFile()) {
+            console.error(red(`No code owner found for "${path}".`));
+            errors++;
+            return false;
+          }
+          return true;
+        })
+        // Get the next level of children for any directories.
+        .map(path => getChildPaths(path)));
+  }
+
+  if (errors) {
+    throw Error(`Found ${errors} files with no owner.`);
+  }
+});
+
+/** Check if the given path is owned by the given owned path matcher. */
+function isOwnedBy(path: string, ownedPath: IMinimatch) {
+  // If the owned path ends with `**` its safe to eliminate whole directories.
+  if (statSync(path).isFile() || ownedPath.pattern.endsWith('**')) {
+    return ownedPath.match('/' + path);
```

Comment:
AFAIK `match(someStr)` ends up creating a Regex under the hood. This could be clearer and faster using `indexOf`.

Reply:
I was originally extracting the regex and working with that, but the `matchBase: true` didn't seem to be coded into the regex. Anyways I think it caches the regex under the hood instead of recompiling it every time, so it should be fine.